java/code/src/com/redhat/rhn/manager/system/SystemManager.java

```diff
@@ -1076,6 +1076,21 @@ public static DataResult<SystemOverview> outOfDateList(User user) {
         return systemListNew(user, PagedSqlQueryBuilder::parseFilterAsNumber, pc);
     }
 
+    /**
+     * Returns the number of systems with outdated packages
+     *
+     * @return number of systems with outdated packages
+     */
+    public static long countOutdatedSystems() {
+        Map<String, Long> params = Collections.emptyMap();
+        SelectMode m = ModeFactory.getMode("System_queries", "count_outdated_systems");
+        DataResult<Map<String, Object>> dr = makeDataResult(params, params, null, m);
+
+        Map<String, Object> result = dr.get(0);
+        Long count = (Long) result.get("count");
+        return count.longValue();
```

Comment:
Given how simple the query is you could probably get away with something like:

```suggestion
        NativeQuery<Tuple> query = HibernateFactory.getSession().createNativeQuery(
            "select count(distinct(id)) as count from susesystemoverview where outdated_packages > 0", Tuple.class);
       return query.getSingleResult().get("count", Number.class).longValue();
```

Reply:
As I mentioned above I'd prefer to keep it consistent with other count queries.